src/clj/game/core/runs.clj

```diff
@@ -125,9 +125,11 @@
              {:prompt (str "Pay " trash-cost "[Credits] to trash " name "?")
               :yes-ability {:cost [:credit trash-cost]
                             :delayed-completion true
-                            :effect (effect (trash eid card nil)
-                                            (system-msg (str "pays " trash-cost
-                                                             " [Credits] to trash " name)))}}}
+                            :effect (req (trash state side eid card nil)
+                                         (let [src (name-zone :corp (:zone card))]
+                                           (system-msg state side (str "pays " trash-cost
+                                                                       " [Credits] to trash " name
+                                                                       (when-not (nil? src) (str " from " src))))))}}}
```

Comment:
Currently it only works for centrals. `name-zone` returns `HQ` for cards in HQ and `HQ Server` for cards install in the root of HQ. Remotes appear to return nil. I don't know if it's right to alter `name-zone`, or whether to do something more custom.